### PR TITLE
Add `cargo check --all`

### PR DESCRIPTION
**Pull request by alexcrichton**
This'll check all `build` targets for all packages in a workspace

### CODE REVIEW

**Comment by crumblingstatue** [comment]
Thanks, this gives symmetry with `cargo build --all`.

File: src/bin/check.rs [hunk 1]
```diff
@@ -14,6 +14,7 @@ Usage:
 Options:
     -h, --help                   Print this message
     -p SPEC, --package SPEC ...  Package to check
+    --all                        Build all packages in the workspace
```
Comment (matklad) [inline src/bin/check.rs]: s/build/check/ ?

File: src/bin/check.rs [hunk 2]
```diff
@@ -14,6 +14,7 @@ Usage:
 Options:
     -h, --help                   Print this message
     -p SPEC, --package SPEC ...  Package to check
```
Comment (matklad) [inline src/bin/check.rs]: There is `...` so both the help text and the flag name should be plural.

**Comment by matklad** [comment]
r+ from me :)

**Pull request merged**